client/src/assets/Search.tsx

```diff
@@ -0,0 +1,58 @@
+import { useState } from "react";
+
+interface SearchProps {
+  username: string;
+  repoName?: string | null;
+  onSearch: (username: string, repo: string) => void;
+}
+
+export default function Search({ username, onSearch, repoName: initialRepoName }: SearchProps) {
+  const [repoName, setRepoName] = useState(initialRepoName ?? "");
+    const handleSearch =() =>{
+        onSearch(username, repoName);
+    }
+    const handleBack = () =>{
+        onSearch(username, "");
+    }
```

Comment:
이벤트 핸들러를 정의하는 방법의 경우 제가 작성한 [이 글](https://junilhwang.github.io/TIL/clean-code/%EC%A1%B0%EA%B0%81%EB%AA%A8%EC%9D%8C/%EC%9D%B4%EB%B2%A4%ED%8A%B8%EC%99%80_%ED%95%A8%EC%88%98%EB%8A%94_%EB%8B%A4%EB%A5%B4%EB%8B%A4/)을 참고해주세요!


이 코드의 경우
```suggestion
export default function Search({ username, onSearch, repoName: initialRepoName }: SearchProps) {
  const [repoName, setRepoName] = useState(initialRepoName ?? "");
  
    const handleSubmit =() =>{
        onSearch(username, repoName);
    }
    const handleBackClick = () =>{
        onSearch(username, "");
    }
```

요로코롬 표현하면 조금 더 자연스러울 것 같아요 ㅎㅎ
- Search라는 행위에 대한 트리거라기보단, Submit 에 대한 트리거
- Back 이라는 행위에 대한 트리거라기보단, 아니라 Back 버튼을 클릭에 대한 트리거